db/decode.go

```diff
@@ -0,0 +1,235 @@
+package db
+
+import (
+	"reflect"
+	"strconv"
+	"time"
+
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+	"github.com/aws/aws-sdk-go/service/dynamodb/dynamodbattribute"
+	"github.com/sul-dlss-labs/taco/datautils"
+)
+
+// A Decoder provides unmarshaling AttributeValues to Go value types.
+type Decoder struct {
+	// Instructs the decoder to decode AttributeValue Numbers as
```

Comment:
i believe this addresses the issues in #395 per @jcoyne's previous comments and review here. Can @jcoyne confirm?